### PR TITLE
Added tests for role and filter hammer commands

### PR DESCRIPTION
**Pull request by svtkachenko**
* Implemented `test_positive_create_with_permission` from `cli/test_rolre` that was stubbed because of bug.
* Adopted `api/filter` tests for cli as hammer `filter` subcommand wasn't tested at all.
* Removed `skip_if_bug_open` for closed bugs.

Tests results:
```python
 py.test -v -k "FilterTestCase or RoleTestCase" tests/foreman/cli/test_{role.py,filter.py}
===================================================================== test session starts =====================================================================
platform linux2 -- Python 2.7.12, pytest-2.9.2, py-1.4.31, pluggy-0.3.1 -- /home/qui/code/robottelo/venv2/bin/python2.7
cachedir: .cache
rootdir: /home/qui/code/robottelo, inifile: 
plugins: cov-2.3.1, xdist-1.14
collected 10 items 

tests/foreman/cli/test_role.py::RoleTestCase::test_positive_create_with_filter PASSED
tests/foreman/cli/test_role.py::RoleTestCase::test_positive_create_with_name PASSED
tests/foreman/cli/test_role.py::RoleTestCase::test_positive_create_with_permissions PASSED
tests/foreman/cli/test_role.py::RoleTestCase::test_positive_delete_by_id PASSED
tests/foreman/cli/test_role.py::RoleTestCase::test_positive_update_name PASSED
tests/foreman/cli/test_filter.py::FilterTestCase::test_positive_create_with_permission PASSED
tests/foreman/cli/test_filter.py::FilterTestCase::test_positive_delete PASSED
tests/foreman/cli/test_filter.py::FilterTestCase::test_positive_delete_role PASSED
tests/foreman/cli/test_filter.py::FilterTestCase::test_positive_update_permisisons PASSED
tests/foreman/cli/test_filter.py::FilterTestCase::test_positive_update_role PASSED

================================================================= 10 passed in 422.58 seconds =================================================================
```

### CODE REVIEW

File: tests/foreman/cli/test_filter.py [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+    def test_positive_update_permisisons(self):
+        """Create a filter and update its permissions.
+
+        @id: 3d6a52d8-2f8f-4f97-a155-9b52888af16e
+
+        @Assert: Permissions updated.
+        """
+        filter_ = make_filter({
+            'role-id': self.role['id'],
+            'permissions': self.perms[0],
```
Comment (oshtaier) [inline tests/foreman/cli/test_filter.py]: Could we use something else, not the indexes? Something more meaningful

File: tests/foreman/cli/test_role.py [hunk 2]
```diff
@@ -43,19 +43,55 @@ def test_positive_create_with_name(self):
...
+    def test_positive_create_with_filter(self):
+        """Create new role with a filter
+
+        @id: 6c99ee25-4e58-496c-af42-f8ad2da6cf07
+
+        @assert: Role is created and correct filter is assigned
+        """
+        role = make_role({'name': gen_string('alpha')})
+        # Pick permissions by its resource type
+        resource_type = 'Organization'
```
Comment (oshtaier) [inline tests/foreman/cli/test_role.py]: Whether put it like a const on a module level or use it as a string directly in the call
We do not use variables for one time usage

**Comment by oshtaier** [comment]
ACK pending comments

File: tests/foreman/cli/test_filter.py [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+from robottelo.decorators import tier1
+from robottelo.test import APITestCase
+
+
+class FilterTestCase(APITestCase):
+
+    @classmethod
+    def setUpClass(cls):
+        """Search for Organization permissions. Set ``cls.perms``.
+        """
```
Comment (abalakh) [inline tests/foreman/cli/test_filter.py]: nitpick: this docstring will fit in one line:
```python
        """Search for Organization permissions. Set ``cls.perms``."""
```

File: tests/foreman/cli/test_filter.py [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+        super(FilterTestCase, cls).setUpClass()
+        cls.perms = [
+            permission['name']
+            for permission in Filter.available_permissions(
+                {'resource-type': 'Organization'})
+            ]
+
+    def setUp(self):
+        self.role = make_role({'name': gen_string('alpha')})
```
Comment (abalakh) [inline tests/foreman/cli/test_filter.py]: Please add a docstring and don't forget to call super

File: tests/foreman/cli/test_filter.py [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+
+        @Assert: The filter cannot be fetched.
+        """
+        filter_ = make_filter({
+            'role-id': self.role['id'],
+            'permissions': self.perms,
+        })
+
+        # A filter depends on a role. Deleting a role implicitly deletes the
+        # filter pointing at it.
```
Comment (abalakh) [inline tests/foreman/cli/test_filter.py]: :+1:

File: tests/foreman/cli/test_role.py [hunk 6]
```diff
@@ -43,19 +43,55 @@ def test_positive_create_with_name(self):
...
-    def test_positive_create_with_permission(self):
+    @tier1
+    def test_positive_create_with_filter(self):
+        """Create new role with a filter
+
+        @id: 6c99ee25-4e58-496c-af42-f8ad2da6cf07
+
+        @assert: Role is created and correct filter is assigned
+        """
+        role = make_role({'name': gen_string('alpha')})
```
Comment (abalakh) [inline tests/foreman/cli/test_role.py]: nitpick: as you don't care about name here, you can call `make_role()` without any arguments.

File: tests/foreman/cli/test_role.py [hunk 7]
```diff
@@ -43,19 +43,55 @@ def test_positive_create_with_name(self):
...
+    def test_positive_create_with_permissions(self):
         """Create new role with a set of permission
 
         @id: 7cb2b2e2-ad4d-41e9-b6b2-c0366eb09b9a
 
         @assert: Role is created and has correct set of permissions
-
-        @caseautomation: notautomated
         """
+        role = make_role({'name': gen_string('alpha')})
```
Comment (abalakh) [inline tests/foreman/cli/test_role.py]: >nitpick: as you don't care about name here, you can call `make_role()` without any arguments.

File: tests/foreman/cli/test_role.py [hunk 8]
```diff
@@ -43,19 +43,55 @@ def test_positive_create_with_name(self):
...
+        # Pick permissions by its resource type
+        resource_type = 'Organization'
```
Comment (abalakh) [inline tests/foreman/cli/test_role.py]: >Whether put it like a const on a module level or use it as a string directly in the call
We do not use variables for one time usage

File: robottelo/cli/factory.py [hunk 9]
```diff
@@ -740,6 +741,53 @@ def make_role(options=None):
...
+    # Role and permissions are required fields.
+    if not options:
+        raise CLIFactoryError('Please provide required parameters')
+
+    # Do we have at least one role field?
+    if not any(options.get(key) for key in ['role', 'role-id']):
+        raise CLIFactoryError('Please provide a valid role field.')
+
+    # Do we have at least one permissions field?
+    if not any(options.get(key) for key in ['permissions', 'permission-ids']):
```
Comment (abalakh) [inline robottelo/cli/factory.py]: So you have checks for role and permissions only. Just curious, is organization optional for filter or not? If it is, then corresponding check is needed too :)

**Comment by abalakh** [comment]
Pending comments

File: robottelo/cli/factory.py (hunk 9 above)
Comment (svtkachenko) [inline robottelo/cli/factory.py]: Yeap, in this case organization looks like optional -- at least it is possible to create filter without setting organization.

File: tests/foreman/cli/test_role.py (hunk 6 above)
Comment (oshtaier) [inline tests/foreman/cli/test_role.py]: if corresponding factory method assign some value for `name` of course

File: tests/foreman/cli/test_role.py (hunk 7 above)
Comment (oshtaier) [inline tests/foreman/cli/test_role.py]: if corresponding factory method assign some value for `name` of course

**Comment by oshtaier** [comment]
APT

File: tests/foreman/cli/test_filter.py [hunk 10]
```diff
@@ -0,0 +1,214 @@
...
+            permission['name']
+            for permission in Filter.available_permissions(
+                {'resource-type': 'User'})
+            ]
+        Filter.update({
+            'id': filter_['id'],
+            'permissions': new_perms
+        })
+        filter_ = Filter.info({'id': filter_['id']})
+        self.assertEqual(filter_['permissions'].split(", "), new_perms)
```
Comment (abalakh) [inline tests/foreman/cli/test_filter.py]: please use sets instead of lists as there's no guarantee elements will be returned in exactly the same order

**Comment by abalakh** [comment]
ACK pending the last comment :)

**Comment by oshtaier** [comment]
Add corresponding entries to robottelo/docs/api/*.rst

**Comment by oshtaier** [comment]
ACK

**Pull request merged**